### ISSUE TITLE
Allow Octicon to take and render alt text

### ISSUE THREAD

**Issue by outofambit** [desktop/desktop#6495]
Our Octicon component doesn't allow setting an alt text (for screen readers, and other accessibility technologies) via its props. Tackling some accessibility bugs requires that we implement this.

According to this somewhat outdated [article](https://css-tricks.com/accessible-svgs/) you can do this by adding a title tag at the beginning of the svg block, which I think will work nicely with our current implementation.

The Octicon component prop should be called `ariaLabel`, as [this is used by the octicons-react package's components](https://github.com/primer/octicons/tree/v8.1.0/lib/octicons_react#arialabel). This will ease our transition whenever we tackle #5185.

The Octicon component's render method lives here:

https://github.com/desktop/desktop/blob/249c06d51332a7254de848e1c77a60e1ae52d587/app/src/ui/octicons/octicon.tsx#L65-L79

**Comment by outofambit** [comment]
We would love for someone from the community to take this on, and if you have interest we're happy to answer questions or point you to the right place in the code base if you aren't able to find it. I'm putting `good first issue` and `help wanted` on this issue to convey that. If you do decide to pick it up, please respond here to indicate you're working on it before submitting a pull request. There will be more accessibility-related issues we'll create over the next couple months, so don't worry if you can't contribute right away. Thanks!

**Comment by shiftkey** [comment]
> According to this somewhat outdated article you can do this by adding a title tag at the beginning of the svg block, which I think will work nicely with our current implementation.

I had a look at what `octicons-react` does internally and it looks like they just use `aria-label` as an attribute.  Does that seem reasonable to address the accessibility issue?

We also already have a `title` prop on `Octicon` that is used to render a tooltip (if defined). This lives inside the `<path>` element inside the `<svg>` element, which is slightly different to what you were proposing here. I guess this means the `title`/`tooltip` thing here is something we'll need to address as part of #5185 because `octicons-react` doesn't have an equivalent currently.

**Comment by outofambit** [comment]
> We also already have a `title` prop on `Octicon` that is used to render a tooltip (if defined).

@shiftkey that's a good point, i hadn't looked at the tool tip thing yet. I think previously we had talked about using #5185 to solve this issue, but I agree it makes more sense to fix this independently and tackle #5185 at a later date.

> I had a look at what `octicons-react` does internally and it looks like they just use `aria-label` as an attribute. Does that seem reasonable to address the accessibility issue?

yep, that's what i was thinking!

**Comment by lisavogtsf** [comment]
@outofambit I would like to pick this up. I still need to study what is currently going on, but I always approve of accessibility improvements! Need alerts anywhere else?

**Comment by lisavogtsf** [comment]
I have made the changes suggested by the css-tricks article linked above, but I note that the SVG 
 rendered by each Octicon has `aria-hidden="true"` set. I believe that will keep any screenreader from picking up on the `<title>` or `<desc>` tags inside the SVG.

I'm not sure how to test the consequences of removing `aria-hidden="true"`, like tons of svg garbage coming out of screenreaders. I gather there are some accessibility testing tools for Electron, but they didn't seem quite what I was looking for: [https://github.com/electron/electron/blob/master/docs/tutorial/accessibility.md](https://github.com/electron/electron/blob/master/docs/tutorial/accessibility.md)

I found an alternative article that recommended a different approach--leaving the svg hidden to aria but adding a visually hidden component that remains available to aria. [https://simplyaccessible.com/article/7-solutions-svgs/](https://simplyaccessible.com/article/7-solutions-svgs/)
```Our most recommended approach when it comes to browser support and consistency across screen readers is to add a visually-hidden element as a sibling element to the <svg>. With this implementation, we’ve found that all browser and screen reader combinations tested were able to announce the link with the expected text announcement.

We also recommend adding aria-hidden=”true” to the <svg> element itself. This is to help prevent having any other text that may be embedded within the SVG be announced by screen readers. Then, the only text that should be announced would be the content within that visually-hidden element.```

I'm going to pause on this for now before putting up a PR (maybe I should just do that anyway? pausing regardless).

**Comment by lisavogtsf** [comment]
Pull request is up, https://github.com/desktop/desktop/pull/6848

I did not take on the suggestion of octicon-react's usage of ariaLabel.

**Comment by shiftkey** [comment]
@lisavogtsf thanks for picking this up. We can talk more about the changes over in #6848! 🎉
